src/rudderstack.ts

```diff
@@ -125,9 +125,9 @@ export class RudderStack {
      *
      * @param curret_page The name or URL of the current page to track the page view event
      */
-    pageView = (current_page: string, platform = 'Deriv App', user_id: string) => {
+    pageView = (current_page: string, platform = 'Deriv App', user_id: string, properties?: {}) => {
         if (this.has_initialized && this.has_identified && current_page !== this.current_page) {
-            this.analytics.page(platform, current_page, { user_id })
+            this.analytics.page(platform, current_page, { user_id }, properties)
```

Comment:
you need to merge the properties in the 3rd parameter

```suggestion
            this.analytics.page(platform, current_page, { user_id, ...properties })
```

Reply:
4th parameter is for `options` if I'm not mistaken.

Reply:
@prince-deriv Actually,the pageView coming from rudderstack is having `properties` as 3rd attribute and 4th one is `options` which we are not using.But we are pushing one more attribute as `userId` into our analytics and then the properties as 4th we can pass it as whatever needed.So ideally while calling the pageview
```
eg:
   Analytics.pageView(window.location.href, {
            loggedIn: true,
            device_type: 'desktop',
        });
```
we are not passing the userId while calling the pageview.
The properties will have custom values which we are passing
<img width="828" alt="Screenshot 2024-10-25 at 11 02 26 AM" src="https://github.com/user-attachments/assets/cce7d72a-b70a-4dee-b821-1bbb146c35b3">

Reply:
Actually I also think @prince-deriv is right.
both of them should be passed in one object.
Check their docs https://www.rudderstack.com/docs/event-spec/standard-events/page/#page-fields for more info.

Reply:
Now i get it, the userID was always outside and on the root level, so now if we change this schema then cro squad also should change their queries too.
So we better to stick with current implementation.